addons/survey/controllers/main.py

```diff
@@ -531,24 +532,29 @@ def survey_submit(self, survey_token, answer_token, **post):
             answer, comment = self._extract_comment_from_answers(question, post.get(str(question.id)))
             errors.update(question.validate_question(answer, comment))
             if not errors.get(question.id):
-                answer_sudo.save_lines(question, answer, comment)
+                answer_sudo.save_lines(question, answer, comment, overwrite=survey_sudo.users_can_go_back)
 
         if errors and not (answer_sudo.survey_time_limit_reached or answer_sudo.question_time_limit_reached):
-            return {'error': 'validation', 'fields': errors}
+            return {}, {'error': 'validation', 'fields': errors}
 
         if not answer_sudo.is_session_answer:
             answer_sudo._clear_inactive_conditional_answers()
 
+        # Get the page questions correct answers if scoring type is scoring after page
+        page_effective_questions = questions - answer_sudo._get_inactive_conditional_questions()
+        correct_answers = survey_sudo.get_correct_answers(page_effective_questions.ids) \
+            if survey_sudo.scoring_type == 'scoring_with_answers_after_page' else {}
```

Comment:
As there is no need for `page_effective_questions` in the other case:
```suggestion
        correct_answers = {}
        if survey_sudo.scoring_type == 'scoring_with_answers_after_page':
            correct_answers = survey_sudo.get_correct_answers(questions - answer_sudo._get_inactive_conditional_questions())
```